x-pack/plugins/observability/server/services/slo/historical_summary_client.ts

```diff
@@ -0,0 +1,302 @@
+
+    return {
+      date: new Date(bucket.key_as_string),
+      errorBudget,
+      sliValue,
+      status: computeSummaryStatus(slo, sliValue, errorBudget),
+    };
+  });
+}
+
+function handleResultForCalendarAlignedAndTimeslices(
+  slo: SLO,
+  buckets: DailyAggBucket[],
+  dateRange: DateRange
+): HistoricalSummary[] {
+  const initialErrorBudget = 1 - slo.objective.target;
+
+  return buckets.map((bucket: DailyAggBucket): HistoricalSummary => {
+    const good = bucket.cumulative_good?.value ?? 0;
+    const total = bucket.cumulative_total?.value ?? 0;
+    const sliValue = computeSLI({ good, total });
+    const totalSlices = computeTotalSlicesFromDateRange(dateRange, slo.objective.timesliceWindow!);
+    const consumedErrorBudget = (total - good) / (totalSlices * initialErrorBudget);
+    const errorBudget = toErrorBudget(initialErrorBudget, consumedErrorBudget);
+
+    return {
+      date: new Date(bucket.key_as_string),
+      errorBudget,
+      sliValue,
+      status: computeSummaryStatus(slo, sliValue, errorBudget),
+    };
+  });
+}
+
+function handleResultForRolling(slo: SLO, buckets: DailyAggBucket[]): HistoricalSummary[] {
+  const initialErrorBudget = 1 - slo.objective.target;
+  const rollingWindowDurationInDays = moment
+    .duration(slo.timeWindow.duration.value, toMomentUnitOfTime(slo.timeWindow.duration.unit))
+    .asDays();
+
+  return buckets
+    .slice(-rollingWindowDurationInDays)
+    .map((bucket: DailyAggBucket): HistoricalSummary => {
+      const good = bucket.cumulative_good?.value ?? 0;
+      const total = bucket.cumulative_total?.value ?? 0;
+      const sliValue = computeSLI({ good, total });
+      const consumedErrorBudget = total === 0 ? 0 : (total - good) / (total * initialErrorBudget);
+      const errorBudget = toErrorBudget(initialErrorBudget, consumedErrorBudget);
+
+      return {
+        date: new Date(bucket.key_as_string),
+        errorBudget,
+        sliValue,
+        status: computeSummaryStatus(slo, sliValue, errorBudget),
+      };
+    });
+}
+
+function generateSearchQuery(slo: SLO, dateRange: DateRange): MsearchMultisearchBody {
+  const unit = toMomentUnitOfTime(slo.timeWindow.duration.unit);
+  const timeWindowDurationInDays = moment.duration(slo.timeWindow.duration.value, unit).asDays();
+
+  return {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          { term: { 'slo.id': slo.id } },
+          { term: { 'slo.revision': slo.revision } },
+          {
+            range: {
+              '@timestamp': {
+                gte: dateRange.from.toISOString(),
+                lte: dateRange.to.toISOString(),
+              },
+            },
+          },
+        ],
+      },
+    },
+    aggs: {
+      daily: {
+        date_histogram: {
+          field: '@timestamp',
+          fixed_interval: '1d',
+          extended_bounds: {
+            min: dateRange.from.toISOString(),
+            max: 'now/d',
+          },
```

Comment:
💬 This forces the number of buckets from the beginning of the period to the current day.